presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeJoinOptimizer.java

```diff
@@ -131,6 +132,11 @@ private boolean isMergeJoinEligible(PlanNode left, PlanNode right, JoinNode node
                 return false;
             }
 
+            // disable merge join optimization when replicated reads is enabled because replicated reads for broadcast join is not supported for grouped execution
```

Comment:
currently implementation for merge join only works for equal-join on bucket key and the join key needs to be in the same join ordering. Eventually it is executed via bucket-to-bucket lifespan by grouped execution which replicated reads cannot support grouped execution because the local execution for the `PerBucket` implementation maintains one `AsyncQueue` for each bucket. Replicated reads strategy cannot create multiple `AsyncQueue` tasks for all buckets which could potential cause `HIVE_EXCEEDED_SPLIT_BUFFERING_LIMIT` issue